src/serial.rs

```diff
@@ -0,0 +1,89 @@
+//! Implementation of [`Serial`](https://docs.rs/embedded-hal/0.2.1/embedded_hal/serial/index.html)
+
+use std::io::{ErrorKind as IoErrorKind, Read, Write};
+
+use nb;
+
+use hal;
+use serial_unix::TTYPort;
+
+/// Newtype around [`serial_unix::TTYPort`] that implements
+/// the `embedded-hal` traits.
+pub struct Serial(pub TTYPort);
```

Comment:
actually, maybe it'd be useful to have a constructor for the wrapped type here?
that way consumers don't also need to import TTYPort.

Reply:
I added the `open` function from `TTYPort`. But if consumers want to do anything other than read/write with the port, like changing the baudrate, they would need to import [`serial_core`](https://docs.rs/serial-core/0.4.0/serial_core/trait.SerialPort.html).